src/odemis/gui/cont/tabs.py

```diff
@@ -7362,8 +7349,7 @@ def _on_specswitch_button_done(self, _):
 
         # check if the state of the buttons need adjustments
         self._adjust_spec_switch_button_state()
-        # cancel the progressive future to update the gauge
-        self._gauge_f.cancel()
+        self._pfc_spec_switch = None
 
         # hide the gauge as there is no convenient way to reset it
         if self.panel.gauge_specswitch.IsShown():
```

Comment:
You can drop this `if` statement. If the gauge is not shown, calling `Hide()` will just do nothing. So it's always fine to call `Hide()`.